donations/sync.py

```
@@ -0,0 +1,205 @@
+        "params": {
+            "tx_hash": tx_hash,
+            "sender_account_id": sender_id,
+            "wait_until": "EXECUTED_OPTIMISTIC",
+        },
+    }
+
+    response = requests.post(rpc_url, json=payload, timeout=30)
+    result = response.json()
+
+    if "error" in result:
+        raise Exception(f"RPC error fetching tx: {result['error']}")
```

Ruff (0.14.14)

[warning] Create your own exception

(TRY002)